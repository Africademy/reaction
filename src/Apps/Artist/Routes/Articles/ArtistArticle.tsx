import { Sans, Serif } from "@artsy/palette"
import React, { SFC } from "react"
import { Box } from "Styleguide/Elements/Box"
import { Flex } from "Styleguide/Elements/Flex"
import { Col, Row } from "Styleguide/Elements/Grid"
import { Image } from "Styleguide/Elements/Image"
import { Separator } from "Styleguide/Elements/Separator"
import { Spacer } from "Styleguide/Elements/Spacer"
import { Responsive } from "Utils/Responsive"

interface ArticleItemProps {
  imageUrl: string
  date: string
  author: string
  title: string
  href: string
}

export const ArticleItem: SFC<ArticleItemProps> = props => {
  return (
    <Responsive>
      {({ xs }) => {
        if (xs) return <SmallArticleItem {...props} />
        else return <LargeArticleItem {...props} />
      }}
    </Responsive>
  )
}

const LargeArticleItem: SFC<ArticleItemProps> = props => {
  const { author, date, href, imageUrl, title } = props

  return (
    <React.Fragment>
      <Row>
        <Col sm={2}>
          <Date size="3" weight="medium">
            {date}
          </Date>
        </Col>
        <Col sm={8}>
          <Flex>
            <Box pr={12}>
              <Title size="5">
                <a href={href} className="noUnderline">
                  {title}
                </a>
              </Title>
              <Credit size="2" color="black60">
                {author}
              </Credit>
            </Box>
          </Flex>
        </Col>
        <Col sm={2}>
          <Image width="135px" height="85px" src={imageUrl} />
        </Col>
      </Row>

      {/* FIXME: Weird block height issue... */}
      <Spacer mt={2} />
      <Separator />
      <Spacer mt={3} />
    </React.Fragment>
  )
}

const SmallArticleItem: SFC<ArticleItemProps> = props => {
  const { author, date, href, imageUrl, title } = props

  return (
    <React.Fragment>
      <Flex justifyContent="space-between">
        <Box pr={3}>
          <Date size="1" weight="medium">
            {date}
          </Date>
          <Spacer mb={0.5} />
          <Title size="2">
            <a href={href} className="noUnderline">
              {title}
            </a>
          </Title>
          <Spacer mb={0.5} />
          <Credit size="1" color="black60">
            {author}
          </Credit>
        </Box>
        <Image width="80px" height="50px" src={imageUrl} />
      </Flex>

      {/* FIXME: Weird block height issue... */}
      <Spacer mt={2} />
      <Separator />
    </React.Fragment>
  )
}

const Date = Sans
const Title = Serif
const Credit = Serif
