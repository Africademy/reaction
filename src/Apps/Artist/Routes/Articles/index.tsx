import { Articles_artist } from "__generated__/Articles_artist.graphql"
import React from "react"
import { createFragmentContainer, graphql } from "react-relay"
import { ArticlesRefetchContainer as Articles } from "./ArticlesRefetchContainer"

export interface ArticlesRouteProps {
  artist: Articles_artist
}

export const ArticlesRoute: React.SFC<ArticlesRouteProps> = props => {
  return <Articles artist={props.artist as any} />
}

export const ArticlesRouteFragmentContainer = createFragmentContainer(
  ArticlesRoute,
  graphql`
    fragment Articles_artist on Artist {
      ...ArticlesRefetchContainer_artist
    }
  `
)
