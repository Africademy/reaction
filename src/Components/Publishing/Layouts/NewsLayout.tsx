import React, { Component } from "react"
import styled from "styled-components"
import { NewsHeadline } from "../News/NewsHeadline"
import { NewsSections } from "../Sections/NewsSections"

interface Props {
  article: any
  expanded?: boolean
}

interface State {
  expanded: boolean
}

export class NewsLayout extends Component<Props, State> {
  state = {
    expanded: this.props.expanded || true,
  }

  render() {
    const { article } = this.props

    return (
      <NewsContainer>
        <NewsHeadline article={article} />
        <NewsSections article={article} />
      </NewsContainer>
    )
  }
}

const NewsContainer = styled.div`
  max-width: 780px;
  margin: auto;
`
