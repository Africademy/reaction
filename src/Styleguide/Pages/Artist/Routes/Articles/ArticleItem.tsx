import { Sans, Serif } from "@artsy/palette"
import React from "react"
import { Box } from "Styleguide/Elements/Box"
import { Flex } from "Styleguide/Elements/Flex"
import { Col, Row } from "Styleguide/Elements/Grid"
import { Image } from "Styleguide/Elements/Image"
import { Responsive } from "Styleguide/Utils/Responsive"

interface ArticleItemProps {
  imageUrl: string
  date: string
  author: string
  title: string
  href: string
}

export const ArticleItem = (props: ArticleItemProps) => {
  return (
    <Responsive>
      {({ xs }) => {
        if (xs) return <SmallArticleItem {...props} />
        else return <LargeArticleItem {...props} />
      }}
    </Responsive>
  )
}

const LargeArticleItem = (props: ArticleItemProps) => {
  return (
    <React.Fragment>
      <Row>
        <Col sm={2}>
          <Date size="3" weight="medium">
            {props.date}
          </Date>
        </Col>
        <Col sm={8}>
          <Flex>
            <Box pr={12}>
              <Title size="5">
                <a href={props.href} className="noUnderline">
                  {props.title}
                </a>
              </Title>
              <Credit size="2" color="black60">
                {props.author}
              </Credit>
            </Box>
          </Flex>
        </Col>
        <Col sm={2}>
          <Image width="135px" height="85px" src={props.imageUrl} />
        </Col>
      </Row>
    </React.Fragment>
  )
}

const SmallArticleItem = (props: ArticleItemProps) => {
  return (
    <React.Fragment>
      <Row>
        <Col sm={2}>
          <Box mb={0.5}>
            <Date size="1" weight="medium">
              {props.date}
            </Date>
          </Box>
        </Col>
        <Col sm={8}>
          <Flex>
            <Box pr={3}>
              <Title size="2">
                <a href={props.href} className="noUnderline">
                  {props.title}
                </a>
              </Title>
              <Credit size="2" color="black60">
                {props.author}
              </Credit>
            </Box>
            <Image width="80px" height="50px" src={props.imageUrl} />
          </Flex>
        </Col>
      </Row>
    </React.Fragment>
  )
}

const Date = Sans
const Title = Serif
const Credit = Serif
