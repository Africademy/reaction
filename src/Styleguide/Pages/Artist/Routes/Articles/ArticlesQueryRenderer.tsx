import { ContextConsumer, ContextProps } from "Components/Artsy"
import React from "react"
import { graphql, QueryRenderer } from "react-relay"
import { ArticlesRefetchContainer } from "./ArticlesRefetchContainer"

interface Props extends ContextProps {
  artistID: string
}

const PAGE_SIZE = 10

export const ArticlesQueryRenderer = ContextConsumer(
  class extends React.Component<Props> {
    render() {
      const { artistID, relayEnvironment } = this.props
      return (
        <QueryRenderer
          environment={relayEnvironment}
          query={graphql`
            query ArticlesQueryRendererQuery($artistID: String!, $first: Int!) {
              artist(id: $artistID) {
                ...ArticlesRefetchContainer_artist @arguments(first: $first)
              }
            }
          `}
          variables={{ artistID, first: PAGE_SIZE }}
          render={({ props }) => {
            if (props) {
              return <ArticlesRefetchContainer artist={props.artist} />
            } else {
              return null
            }
          }}
        />
      )
    }
  }
)
