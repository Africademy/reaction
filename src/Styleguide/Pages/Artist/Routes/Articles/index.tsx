import React from "react"
import { ArticlesRefetchContainer } from "./ArticlesRefetchContainer"

interface Props {
  artist: any
}

export class ArticlesRoute extends React.Component<Props> {
  render() {
    return <ArticlesRefetchContainer artist={this.props.artist} />
  }
}
