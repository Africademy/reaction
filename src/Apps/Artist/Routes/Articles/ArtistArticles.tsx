import { ArtistArticles_artist } from "__generated__/ArtistArticles_artist.graphql"
import React, { Component } from "react"
import { createRefetchContainer, graphql, RelayRefetchProp } from "react-relay"
import { PaginationFragmentContainer as Pagination } from "Styleguide/Components/Pagination"
import { Flex } from "Styleguide/Elements/Flex"
import { Col, Row } from "Styleguide/Elements/Grid"
import { ArticleItem } from "./ArtistArticle"

import {
  LoadingArea,
  LoadingAreaState,
} from "Apps/Artist/Components/LoadingArea"

const PAGE_SIZE = 10

interface ArtistArticlesProps {
  relay: RelayRefetchProp
  artist: ArtistArticles_artist
}

export class ArtistArticles extends Component<
  ArtistArticlesProps,
  LoadingAreaState
> {
  state = {
    isLoading: false,
  }

  loadNext = () => {
    const {
      artist: {
        articlesConnection: {
          pageInfo: { hasNextPage, endCursor },
        },
      },
    } = this.props

    if (hasNextPage) {
      this.loadAfter(endCursor)
    }
  }

  loadAfter = cursor => {
    this.toggleLoading(true)

    this.props.relay.refetch(
      {
        first: PAGE_SIZE,
        after: cursor,
        artistID: this.props.artist.id,
        before: null,
        last: null,
      },
      null,
      error => {
        this.toggleLoading(false)

        if (error) {
          console.error(error)
        }
      }
    )
  }

  toggleLoading = isLoading => {
    this.setState({
      isLoading,
    })
  }

  render() {
    return (
      <React.Fragment>
        <span id="jumpto--artistArticles" />

        <Row>
          <Col>
            <LoadingArea isLoading={this.state.isLoading}>
              {this.props.artist.articlesConnection.edges.map(
                ({ node }, index) => {
                  return (
                    <ArticleItem
                      title={node.thumbnail_title}
                      imageUrl={node.thumbnail_image.resized.url}
                      date={node.published_at}
                      author={node.author.name}
                      href={node.href}
                      key={index}
                    />
                  )
                }
              )}
            </LoadingArea>
          </Col>
        </Row>
        <Row>
          <Col>
            <Flex mb={2} justifyContent="flex-end">
              <Pagination
                pageCursors={
                  this.props.artist.articlesConnection.pageCursors as any
                }
                onClick={this.loadAfter}
                onNext={this.loadNext}
                scrollTo="#jumpto-RouteTabs"
              />
            </Flex>
          </Col>
        </Row>
      </React.Fragment>
    )
  }
}

export const ArtistArticlesRefetchContainer = createRefetchContainer(
  ArtistArticles,
  {
    artist: graphql`
      fragment ArtistArticles_artist on Artist
        @argumentDefinitions(
          first: { type: "Int", defaultValue: 10 }
          last: { type: "Int" }
          after: { type: "String" }
          before: { type: "String" }
        ) {
        id
        articlesConnection(
          first: $first
          after: $after
          before: $before
          last: $last
          sort: PUBLISHED_AT_DESC
          in_editorial_feed: true
        ) {
          pageInfo {
            hasNextPage
            endCursor
          }
          pageCursors {
            ...Pagination_pageCursors
          }
          edges {
            node {
              href
              thumbnail_title
              author {
                name
              }
              published_at(format: "MMM d, YYYY")
              thumbnail_image {
                resized(width: 300) {
                  url
                }
              }
              href
            }
          }
        }
      }
    `,
  },
  graphql`
    query ArtistArticlesQuery(
      $first: Int
      $last: Int
      $after: String
      $before: String
      $artistID: String!
    ) {
      artist(id: $artistID) {
        ...ArtistArticles_artist
          @arguments(first: $first, last: $last, after: $after, before: $before)
      }
    }
  `
)
