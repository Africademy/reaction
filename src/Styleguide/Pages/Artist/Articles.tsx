import React from "react"
import {
  graphql,
  QueryRenderer,
  createRefetchContainer,
  RelayRefetchProp,
} from "react-relay"
import { ContextProps, ContextConsumer } from "Components/Artsy"
import styled from "styled-components"
import { Col, Row } from "Styleguide/Elements/Grid"
import { Box } from "Styleguide/Elements/Box"
import { Flex } from "Styleguide/Elements/Flex"
import { Separator } from "Styleguide/Elements/Separator"
import { ArticleItem } from "./ArticleItem"
import { Pagination } from "Styleguide/Components/Pagination"

import { Articles_artist } from "__generated__/Articles_artist.graphql"

interface ArticlesProps {
  relay: RelayRefetchProp
  artist: Articles_artist
}

const PAGE_SIZE = 10

// Renders list of articles, we'll use this in our refetch container.
class ArticlesContainer extends React.Component<ArticlesProps> {
  loadPrev = () => {
    const {
      startCursor,
      hasPreviousPage,
    } = this.props.artist.articlesConnection.pageInfo
    if (hasPreviousPage) {
      this.loadBefore(startCursor)
    }
  }

  loadNext = () => {
    const {
      hasNextPage,
      endCursor,
    } = this.props.artist.articlesConnection.pageInfo
    if (hasNextPage) {
      this.loadAfter(endCursor)
    }
  }

  loadBefore(cursor) {
    this.props.relay.refetch(
      {
        first: null,
        before: cursor,
        artistID: this.props.artist.id,
        after: null,
        last: PAGE_SIZE,
      },
      null,
      error => {
        if (error) {
          console.error(error)
        }
      }
    )
  }

  loadAfter = cursor => {
    this.props.relay.refetch(
      {
        first: PAGE_SIZE,
        after: cursor,
        artistID: this.props.artist.id,
        before: null,
        last: null,
      },
      null,
      error => {
        if (error) {
          console.error(error)
        }
      }
    )
  }

  renderPagination() {
    return (
      <div>
        <Pagination
          {...this.props.artist.articlesConnection.pageCursors}
          onClick={this.loadAfter}
          onNext={this.loadNext}
          onPrev={this.loadPrev}
        />
      </div>
    )
  }

  render() {
    return (
      <div>
        <Row>
          <Col>
            <ArticleList>
              {this.props.artist.articlesConnection.edges.map(({ node }) => {
                return (
                  <ArticleItem
                    title={node.thumbnail_title}
                    imageUrl={node.thumbnail_image.resized.url}
                    date={node.published_at}
                    author={node.author.name}
                    href={node.href}
                  />
                )
              })}
            </ArticleList>
          </Col>
        </Row>

        <Box my={2}>
          <Separator />
        </Box>

        <Row>
          <Col>
            <Flex mb={2} justifyContent="flex-end">
              {this.renderPagination()}
            </Flex>
          </Col>
        </Row>
      </div>
    )
  }
}

// This is the actual Refetch Container we want to use.
const ArticlesRefetchContainer = createRefetchContainer(
  ArticlesContainer,
  {
    artist: graphql`
      fragment Articles_artist on Artist
        @argumentDefinitions(
          first: { type: "Int" }
          last: { type: "Int" }
          after: { type: "String" }
          before: { type: "String" }
        ) {
        id
        articlesConnection(
          first: $first
          after: $after
          before: $before
          last: $last
          sort: PUBLISHED_AT_DESC
          in_editorial_feed: true
        ) {
          pageInfo {
            hasNextPage
            hasPreviousPage
            startCursor
            endCursor
          }
          pageCursors {
            around {
              cursor
              page
              isCurrent
            }
            first {
              cursor
              page
              isCurrent
            }
            last {
              cursor
              page
              isCurrent
            }
          }
          edges {
            node {
              href
              thumbnail_title
              author {
                name
              }
              published_at(format: "MMM d, YYYY")
              thumbnail_image {
                resized(width: 300) {
                  url
                }
              }
              href
            }
          }
        }
      }
    `,
  },
  graphql`
    query ArticlesQuery(
      $first: Int
      $last: Int
      $after: String
      $before: String
      $artistID: String!
    ) {
      artist(id: $artistID) {
        ...Articles_artist
          @arguments(first: $first, last: $last, after: $after, before: $before)
      }
    }
  `
)

interface Props extends ContextProps {
  artistID: string
}

class ArticlesQueryRenderer extends React.Component<Props> {
  render() {
    const { artistID, relayEnvironment } = this.props
    return (
      <QueryRenderer
        environment={relayEnvironment}
        query={graphql`
          query ArticlesIndexQuery($artistID: String!, $first: Int!) {
            artist(id: $artistID) {
              ...Articles_artist @arguments(first: $first)
            }
          }
        `}
        variables={{ artistID, first: PAGE_SIZE }}
        render={({ props }) => {
          if (props) {
            return <ArticlesRefetchContainer artist={props.artist} />
          } else {
            return null
          }
        }}
      />
    )
  }
}

export const ArticlesContent = ContextConsumer(ArticlesQueryRenderer)

const ArticleList = styled.div``
