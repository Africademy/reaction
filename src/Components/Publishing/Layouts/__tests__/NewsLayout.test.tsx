import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { NewsArticle } from "../../Fixtures/Articles"
import { NewsLayout } from "../NewsLayout"

it("renders the news layout properly", () => {
  const series = renderer.create(<NewsLayout article={NewsArticle} />).toJSON()
  expect(series).toMatchSnapshot()
})
