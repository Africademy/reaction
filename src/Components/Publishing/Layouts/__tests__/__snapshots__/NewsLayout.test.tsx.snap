// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the news layout properly 1`] = `
.c1 {
  margin: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: left;
  max-width: 780px;
  width: 100%;
  margin: 40px auto;
  box-sizing: border-box;
  margin-bottom: 30px;
}

.c3 {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 34px;
  line-height: 1.1em;
  font-weight: 600;
}

.c6 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.c4 {
  box-sizing: border-box;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding-right: 0.5rem;
  padding-left: 0.5rem;
}

.c9 {
  position: relative;
}

.c9:hover .file__Fullscreen-yu39to-0 {
  opacity: 1;
}

.c9 .BlockImage__container {
  opacity: 0;
  -webkit-transition: opacity 1s;
  transition: opacity 1s;
}

.c9 .image-loaded {
  opacity: 1;
}

.c10 {
  display: block;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 10px 0 10px 0;
}

.c12 {
  padding: 0;
  width: 100%;
  word-break: break-word;
}

.c12 > p,
.c12 p,
.c12 .public-DraftEditorPlaceholder-root,
.c12 .public-DraftStyleDefault-block {
  font-family: Unica77LLWebRegular , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 14px;
  line-height: 1.4em;
  color: #999;
  margin: 0;
}

.c12 > a,
.c12 a {
  color: #999;
}

.c12 > a:hover,
.c12 a:hover {
  color: black;
}

.c8 {
  margin-right: 0px;
  width: 100%;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c14 {
  position: relative;
  width: 100%;
}

.c14 a {
  color: black;
  text-decoration: none;
  position: relative;
  background-image: linear-gradient(to bottom,transparent 0,#333 1px,transparent 0);
  background-size: 1.25px 4px;
  background-repeat: repeat-x;
  background-position: bottom;
}

.c14 a:hover {
  color: #999;
  opacity: 1;
}

.c14 p,
.c14 ul,
.c14 ol,
.c14 div[data-block=true] .public-DraftStyleDefault-block {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 23px;
  line-height: 1.5em;
  padding-top: 1em;
  padding-bottom: 1em;
  margin: 0;
  font-style: inherit;
}

.c14 p:first-child,
.c14 div[data-block=true]:first-child .public-DraftStyleDefault-block {
  padding-top: 0;
}

.c14 p:last-child,
.c14 div[data-block=true]:last-child .public-DraftStyleDefault-block {
  padding-bottom: 0;
}

.c14 ul,
.c14 ol {
  padding-left: 1em;
}

.c14 li {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 23px;
  line-height: 1.5em;
  padding-top: .5em;
  padding-bottom: .5em;
}

.c14 h1 {
  font-family: Unica77LLWebRegular , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 40px;
  line-height: 1.1em;
  font-weight: normal;
  padding-top: 107px;
  padding-bottom: 46px;
  margin: 0;
  position: relative;
  text-align: center;
}

.c14 h1:before {
  content: "";
  width: 15px;
  height: 15px;
  background: black;
  border-radius: 50%;
  position: absolute;
  top: 69px;
  right: calc(50% - 7.5px);
}

.c14 h2 {
  font-family: Unica77LLWebRegular , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 32px;
  line-height: 1.1em;
  font-weight: normal;
  margin: 0;
}

.c14 h2 a {
  background-size: 1.25px 1px;
}

.c14 h3 {
  font-family: Unica77LLWebRegular , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
  font-weight: normal;
  padding-top: 23px;
  margin: 0;
}

.c14 h3 strong {
  font-weight: normal;
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
}

.c14 h3 a {
  background-size: 1.25px 1px;
}

.c14 blockquote {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
  text-align: left;
  font-weight: normal;
  padding-top: 10px;
  padding-bottom: 10px;
  margin: 0 50px;
  word-break: break-word;
}

.c14 blockquote a {
  background-size: 1.25px 1px;
}

.c14 .content-end {
  display: inline-block;
  content: "";
  width: 12px;
  height: 12px;
  background: black;
  border-radius: 50%;
  margin-left: 12px;
}

.c14 .artist-follow {
  vertical-align: middle;
  margin-left: 10px;
  cursor: pointer;
  background: none transparent;
}

.c14 .artist-follow:before {
  font-family: "artsy-icons";
  content: "";
  vertical-align: -8.5px;
  line-height: 32px;
  font-size: 32px;
}

.c14 .artist-follow:after {
  content: "Follow";
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 17px;
  line-height: 1.1em;
  text-transform: none;
}

.c5 {
  margin-bottom: 20px;
}

.c13 {
  max-width: 660px;
  margin-bottom: 20px;
}

.c0 {
  max-width: 780px;
  margin: auto;
}

@media (max-width:720px) {
  .c1 {
    margin: 0 20px;
  }
}

@media (max-width:720px) {
  .c3 {
    font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 23px;
    line-height: 1.5em;
    line-height: 1.1;
  }
}

@media (max-width:720px) {
  .c9 .file__Fullscreen-yu39to-0 {
    opacity: 1;
  }
}

@media (max-width:600px) {
  .c11 {
    padding: 0px 10px;
  }
}

@media (max-width:600px) {
  .c12 {
    padding: 0px;
  }
}

@media (max-width:600px) {
  .c8 {
    margin-bottom: 10px;
  }
}

@media (max-width:600px) {
  .c7 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:600px) {
  .c14 p,
  .c14 ul,
  .c14 ol,
  .c14 div[data-block=true] .public-DraftStyleDefault-block {
    font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 19px;
    line-height: 1.5em;
  }

  .c14 li {
    font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 19px;
    line-height: 1.5em;
  }

  .c14 h1 {
    font-family: Unica77LLWebRegular , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 34px;
    line-height: 1.1em;
  }

  .c14 h2 {
    font-family: Unica77LLWebRegular , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 32px;
    line-height: 1.1em;
  }

  .c14 h3 {
    font-family: Unica77LLWebRegular , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
    line-height: 1.5em;
  }

  .c14 h3 strong {
    font-family: Unica77LLWebMedium , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
  }

  .c14 blockquote {
    font-family: Unica77LLWebMedium , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
    margin: 0 10px;
  }

  .c14 .content-start {
    font-size: 55px;
  }
}

@media (max-width:720px) {
  .c5 {
    margin: 0 20px 15px 20px;
    padding: 0px;
  }
}

@media (max-width:720px) {
  .c13 {
    margin: 0 20px 15px 20px;
    padding: 0px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        The oldest known depiction of a supernova was found in a work of 5,000 year old rock art, scientists believe.
      </div>
    </div>
  </div>
  <div
    className="c4"
  >
    <div
      className="c5 c6"
      type="image_collection"
    >
      <div
        className="c7"
      >
        <div
          className="c8"
          width={undefined}
        >
          <div
            className="article-image"
          >
            <div
              className="c9"
            >
              <img
                alt="Illustration by Tomi Um for Artsy."
                className="BlockImage__container c10"
                height="auto"
                src="https://d7hftxdivxxvm.cloudfront.net?resize_to=width&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FN13JE8AbtWFAgvueH8G1uQ%2Flarger.jpg&width=1200&quality=80"
                width="100%"
              />
            </div>
            <div
              className="c11"
            >
              <div
                className="c12"
              >
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<p>Illustration by Tomi Um for Artsy.</p>",
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="c13 c6"
      type="text"
    >
      <div
        className="article__text-section c14"
        color="black"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p><strong>The design for the as-yet-uncompleted sculpture</strong>, <span style='text-decoration:line-through;'><em>Bouquet of Tulips</em></span>, was donated by Koons to the French capital in November 2016 as a memorial to the recent terrorist attacks that have taken place in the city. But Koons, <a href='#'>one of the world’s richest living artists</a>, didn’t donate the $4.3 million needed to create 40-foot-tall sculpture, which was raised separately via donations. And the work, slated for installation in front of the Palais de Tokyo and Paris’s Museum of Modern Art, has attracted opposition. In a letter published in the French newspaper Libération on Sunday, signatories—including Frédéric Mitterrand, the country’s former culture minister—demanded that the city halt its plans to install the sculpture, calling it “shocking.” </p>",
            }
          }
        />
      </div>
    </div>
    <div
      className="c13 c6"
      type="text"
    >
      <div
        className="article__text-section c14"
        color="black"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<blockquote>It is understood the [British Museum] has been in talks about a possible loan of the tapestry for several years, but there will be other contenders to host it.</blockquote>",
            }
          }
        />
      </div>
    </div>
    <div
      className="c13 c6"
      type="text"
    >
      <div
        className="article__text-section c14"
        color="black"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<blockquote>The bookmaker Ladbrokes is offering odds on where it might go, with the British Museum evens favourite, followed by Westminster Cathedral at 3/1, Canterbury Cathedral at 5/1 and Hastings at 8/1.</blockquote>",
            }
          }
        />
      </div>
    </div>
    <div
      className="c13 c6"
      type="text"
    >
      <div
        className="article__text-section c14"
        color="black"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<blockquote><a href='#'>The Guardian</a></blockquote>",
            }
          }
        />
      </div>
    </div>
    <div
      className="c13 c6"
      type="text"
    >
      <div
        className="article__text-section c14"
        color="black"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>It further criticized Koons for using the opportunity as a publicity stunt, as the sculpture’s planned location is in a tourist-heavy area far from where the 2015 terrorist attacks actually took place. “We appreciate gifts, [but ones that are] free, unconditional, and without ulterior motives,” the letter said. In any case, Parisian officials have not yet granted authorization to install the sculpture, according to the <em>New York Times</em>.</p>",
            }
          }
        />
      </div>
    </div>
    <div
      className="c13 c6"
      type="text"
    >
      <div
        className="article__text-section c14"
        color="black"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<h3><strong>Takeaway</strong></h3><h3>Should Prince lose at trial and on appeal the resulting precedent would reign in the broader <em>fair use interpretations</em> now afforded to artists.</h3>",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
